public/edit-task.js

```diff
@@ -0,0 +1,67 @@
+const taskIDDOM = document.querySelector('.task-edit-id')
+const taskNameDOM = document.querySelector('.task-edit-name')
+const taskCompletedDOM = document.querySelector('.task-edit-completed')
+const editFormDOM = document.querySelector('.single-task-form')
+const editBtnDOM = document.querySelector('.task-edit-btn')
+const formAlertDOM = document.querySelector('.form-alert')
+const params = window.location.search
+const id = new URLSearchParams(params).get('id')
+let tempName
+
+const showTask = async () => {
+  try {
+    const {
+      data: { task },
+    } = await axios.get(`/api/v1/tasks/${id}`)
+    const { _id: taskID, completed, name } = task
+
+    taskIDDOM.textContent = taskID
+    taskNameDOM.value = name
+    tempName = name
+    if (completed) {
+      taskCompletedDOM.checked = true
+    }
+  } catch (error) {
+    console.log(error)
+  }
+}
+
+showTask()
+
+editFormDOM.addEventListener('submit', async (e) => {
+  editBtnDOM.textContent = 'Loading...'
+  e.preventDefault()
+  try {
+    const taskName = taskNameDOM.value
+    const taskCompleted = taskCompletedDOM.checked
```

Comment:
Again, it's fine for you first time that you use DOM, and mark the params according tot he given data, but please keep in you mind that in a real fronend framework all of this will be look different.